ivy_tests/test_ivy/helpers/function_testing.py

```diff
@@ -365,21 +370,46 @@ def test_function(
         and not instance_method
         and "bool" not in input_dtypes
     ):
-        gradient_test(
-            fn_name=fn_name,
-            all_as_kwargs_np=all_as_kwargs_np,
-            args_np=args_np,
-            kwargs_np=kwargs_np,
-            input_dtypes=input_dtypes,
-            as_variable_flags=as_variable_flags,
-            native_array_flags=native_array_flags,
-            container_flags=container_flags,
-            rtol_=rtol_,
-            atol_=atol_,
-            xs_grad_idxs=xs_grad_idxs,
-            ret_grad_idxs=ret_grad_idxs,
-            ground_truth_backend=ground_truth_backend,
-        )
+        if hasattr_unsupported_gradients and fw in fw_list:
+            if ivy.nested_argwhere(
+                all_as_kwargs_np,
+                lambda x: x.dtype in fw_list[fw] if isinstance(x, np.ndarray) else None,
```

Comment:
I think this line could be an issue, we should instead coompare the `input_dtypes` to the `fw_list[fw]` as they are guaranteed to be strings unlike `x.dtype`.

Reply:
@VedPatwardhan if x is an instance of np.ndarray then x.dtype must be a string right?